test/fastly/client_test.rb

```diff
@@ -20,20 +20,17 @@
       assert_equal nil, client.password
     end
 
+    it 'raises an Unauthorized if api_key is not passed in the options' do
+      assert_raises(Fastly::Unauthorized) { Fastly::Client.new(user: user, password: password)}
+    end
+
     it 'raises Unauthorized if user/pass provided but are invalid' do
       stub_request(:any, /api.fastly.com/).to_return(status: 400)
 
       e = assert_raises(Fastly::Unauthorized) {
-        Fastly::Client.new(user: user, password: pass)
+        Fastly::Client.new(user: user, password: password)
```

Comment:
Was this assertion always failing? As I don't see `pass` set anywhere (it looks like the variable was always named `password`).

Reply:
I've not written Ruby in many years but a quick reduced test case would suggest this should have been a syntax error. Something along the lines of:

```
undefined local variable or method `pass'
```

Reply:
You're right. I think these were passing before because it was stubbing an `Unauthorized` error. The error that surfaced wasn't from the undefined variable so I'm assuming it didn't matter that the `pass` wasn't being set anywhere. However, I think I'm going to fix all of them to reference `password` because it'll be clear to everyone what the errors should be.

Reply:
This is a bit more problematic: `pass` is a [minitest assertion](https://docs.seattlerb.org/minitest/Minitest/Assertions.html#method-i-pass):

```
# File lib/minitest/assertions.rb, line 622
def pass _msg = nil
  assert true
end
```

Here's an example 
```ruby
# foo_test.rb                                                                                     
require 'minitest/autorun'

class FooTest < Minitest::Test
  def test_pass
    pass
  end

  def test_nothing
  end
end
```
```
$ ruby foo_test.rb                                               
Run options: --seed 55807

# Running:

..

Finished in 0.000752s, 2659.5740 runs/s, 1329.7870 assertions/s.

2 runs, 1 assertions, 0 failures, 0 errors, 0 skips
```

An empty test produces no assertions, but pass creates one assertion which always passes. 

Changing this to `password` seems like a good idea.

Reply:
Thanks @stevendaniels for that confirmation!